Scenarios/run_Scenario.bat

```diff
@@ -1,74 +1,211 @@
-ECHO SYSTEM INFO.
+SETLOCAL ENABLEDELAYEDEXPANSION
+
 ECHO ====================================
-setlocal EnableDelayedExpansion
+ECHO ####################################
+ECHO #       SIMULATION TOOL SELECTION         #
+ECHO ####################################
+
+SET sel=0
+ECHO Please select simulation tool:
+ECHO [0]: Abort program
+ECHO [1]: Esmini
+ECHO [2]: openPASS
+SET /p sel="Enter ID: "
+
+IF %sel% LSS 1 GOTO finish
+IF %sel% GTR 2 GOTO finish
+IF %sel% EQU 1 GOTO Esmini
+IF %sel% EQU 2 GOTO OpenPASS
+
+:Esmini
+
+ECHO ####################################
+ECHO #    ESMINI: SYSTEM SETTINGS       #
+ECHO ####################################
 
-if not defined ESMINI (
+IF NOT DEFINED ESMINI (
     ECHO Esmini path not found. Please install and create ESMINI environment variable.
-    pause
+    PAUSE
     GOTO finish
 )
 
 ECHO esmini path: %ESMINI%
 
-for /F "tokens=2 delims==" %%a in ('findstr /I "ESMINI_GIT_TAG=" "%ESMINI%\..\version.txt"') do set "ReleaseVersion=%%a"
-set ReleaseVersion=!ReleaseVersion:~2,-1!
+FOR /F "tokens=2 delims==" %%a in ('findstr /I "ESMINI_GIT_TAG=" "%ESMINI%\..\version.txt"') DO SET "ReleaseVersion=%%a"
+SET ReleaseVersion=!ReleaseVersion:~2,-1!
 
-for /F "tokens=1,2,3 delims=." %%a in ("!ReleaseVersion!") do (
-   set Major=%%a
-   set Minor=%%b
-   set Revision=%%c
+FOR /F "tokens=1,2,3 delims=." %%a in ("!ReleaseVersion!") DO (
+    SET Major=%%a
+    SET Minor=%%b
+    SET Revision=%%c
 )
 
-echo esmini version is: !Major!.!Minor!.!Revision!
+ECHO esmini version is: !Major!.!Minor!.!Revision!
 
-set supported=1
+SET supported=1
 
-if !Major! LSS 1 set supported=0
-if !Major! EQU 1 (
-    if !Minor! LSS 7 set supported=0
-    if !Minor! EQU 7 (
-        if !Revision! LSS 13 set supported=0
+IF !Major! LSS 1 SET supported=0
+IF !Major! EQU 1 (
+    IF !Minor! LSS 7 SET supported=0
+    IF !Minor! EQU 7 (
+        IF !Revision! LSS 13 SET supported=0
         )
     )
 )
 
-if !supported! EQU 0 (
-    echo Unsupported esmini version.
-    goto finish
+IF !supported! EQU 0 (
+    ECHO Unsupported esmini version.
+    GOTO finish
 )
 
-:: Section 2: Parse OpenScenario files in same folder.
-set x=0
-for /f "delims=" %%a in ('dir /s /b *.xosc') do (
-   set /A "x+=1"
-   set list[!x!]=%%a
+SET x=0
+FOR /f "delims=" %%a in ('dir /s /b *.xosc') DO (
+    SET /A "x+=1"
+    SET list[!x!]=%%a
 )
-set Filesx=%x%
+SET Filesx=%x%
 
-:again
-ECHO ====================================
-ECHO List of OpenSCENARIO (XOSC) files.
-ECHO ====================================
-rem Display array elements
-for /L %%i in (1,1,%Filesx%) do echo [%%i]: "!list[%%i]!"
-ECHO ====================================
-ECHO Select and play a scenario by number.
+ECHO ####################################
+ECHO #       ESMINI: SIMULATION         #
+ECHO ####################################
+
+ECHO Supported OpenSCENARIO (XOSC) files:
+FOR /L %%i in (1,1,%Filesx%) DO ECHO [%%i]: "!list[%%i]!"
+:again_esmini
```

Comment:
When now executing multiple scenarios subsequently with esmini you don´t get the list of available scenarios shown again. Can we move the :again_esmini two lines up please?

Reply:
Fixed in 737f37a60d913b88b011e5d4120ee9e5582d2e59